### ISSUE TITLE
Contribute upstream to ansible-modules-extras

### ISSUE THREAD

**Issue by brianhelba** [UDST/ansible-conda#1]
This seems like it would be a worthy addition to upstream Ansible. Would you consider submitting this to https://github.com/ansible/ansible-modules-extras ?

If you're not interested in dealing with that process, I'd be happy to submit a PR to ansible-modules-extras myself (maintaining credit to you as Git author and copyright owner).

**Comment by jiffyclub** [comment]
I'll have to talk to my supervisors about this, but I'm 100% for it. Do you think the module needs to wrap up anymore conda functionality like environments?

**Comment by brianhelba** [comment]
@jiffyclub Good point. I imagine that the module could be merged as-is, but it's going to much more difficult to maintain once it's merged into an upstream repo.

I'll ask some of the other Anaconda users in my organization to see if there are any feature requests that we could add now.

**Comment by brianhelba** [comment]
@jiffyclub Also, when the PR to ansible-modules-extras is made, I'd prefer it be made in a way that preserves the full development history from here. It's not that difficult to graft this repo's history into the upstream one, so once we proceed, I can make the new branch in ansible-modules-extras, but you can issue the actual PR, if you'd like.

**Comment by jiffyclub** [comment]
I've just come upon a situation with a shared web server where I think it would be nice to be able to manage conda environments via Ansible. So it might be worthwhile keeping control of this for now.

**Comment by koverholt** [comment]
+1, I'll pass this along to the Anaconda Cluster dev team

**Comment by brianhelba** [comment]
I just talked to @koverholt at SciPy today. He suggested that we also include support for fetching packages from custom channels.

Once the channels feature is added, and we ensure that environments are working properly, I think we'll be ready to submit this upstream.

**Comment by brianhelba** [comment]
Also, apparently Conda already has support in Salt. We should probably look at that and see if there are any features we want to ~~shamelessly copy~~ implement for ourselves.

**Comment by maxalbert** [comment]
@jiffyclub The current version does in fact allow managing environments by passing the `--name` argument in `extra_args` (at least with the fix in the PR #4 that I just submitted). But I agree it would be nice to have more "high-level" support for environments via a proper option in this module.

**Comment by lowks** [comment]
+1 Has this been submitted upstream ?

**Comment by jiffyclub** [comment]
@lowks it has not.

**Comment by lowks** [comment]
Oh. Any reason why not ?

**Comment by jiffyclub** [comment]
No particularly good reason, I think. If other people on this thread (particularly @brianhelba) are still interested and think it's ready we should see about it. But I've since changed jobs and I no longer actively work on this or have ultimate control.

**Comment by lowks** [comment]
Right. @brianhelba ??

**Comment by lowks** [comment]
Sent for PR. https://github.com/ansible/ansible-modules-core/pull/3780

**Comment by RafalSkolasinski** [comment]
Is there any work going on this PR?

**Comment by tmoschou** [comment]
Hi guys, I've created a PR for a conda module upstream https://github.com/ansible/ansible/pull/40455
